crrExistingObjects.js

```diff
@@ -17,16 +17,20 @@ let STORAGE_TYPE = process.env.STORAGE_TYPE;
 let TARGET_REPLICATION_STATUS = process.env.TARGET_REPLICATION_STATUS;
 const TARGET_PREFIX = process.env.TARGET_PREFIX;
 const WORKERS = (process.env.WORKERS &&
-                 Number.parseInt(process.env.WORKERS, 10)) || 10;
+    Number.parseInt(process.env.WORKERS, 10)) || 10;
 const MAX_UPDATES = (process.env.MAX_UPDATES &&
-                     Number.parseInt(process.env.MAX_UPDATES, 10));
+    Number.parseInt(process.env.MAX_UPDATES, 10));
 const MAX_SCANNED = (process.env.MAX_SCANNED &&
-                     Number.parseInt(process.env.MAX_SCANNED, 10));
+    Number.parseInt(process.env.MAX_SCANNED, 10));
 let KEY_MARKER = process.env.KEY_MARKER;
 let VERSION_ID_MARKER = process.env.VERSION_ID_MARKER;
+const RETRY_ON_ERROR = (process.env.RETRY_ON_ERROR &&
```

Comment:
I think if we want to leave a choice to the user to tune the retry behavior, it's better to default to the most robust approach (retry x times or a certain amount of time) and possibly have a tunable to disable retries or configure their number. I'm not convinced it's very useful to offer this tunable here if we provide a reasonable default behavior.

Reply:
Removed `RETRY_ON_ERROR` and as suggested, s3 requests on failure will be retried 100 (a set constant) using the aws sdk's native feature. `customBackOff` is modified to try out these 100 requests over a span of 5 minutes (computed constant). Recursion is also removed. So the errors will eventually stop the process after 100 retries.